tests/sles4sap/hana_cluster.pm

```diff
@@ -0,0 +1,103 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Configure HANA-SR cluster
+# Maintainer: Ricardo Branco <rbranco@suse.de>
+
+use base "sles4sap";
+use testapi;
+use lockapi;
+use hacluster;
+use utils 'systemctl';
+use strict;
+use warnings;
+
+sub run {
+    my ($self)       = @_;
+    my $instance_id  = get_required_var('INSTANCE_ID');
+    my $sid          = get_required_var('INSTANCE_SID');
+    my $cluster_name = get_cluster_name;
+
+    # Set SAP variables
+    my $pscmd  = $self->set_ps_cmd("HDB");
+    my $sapadm = $self->set_sap_info($sid, $instance_id);
+
+    # Synchronize the nodes
+    barrier_wait "HANA_CLUSTER_INSTALL_$cluster_name";
+
+    $self->select_serial_terminal;
+
+    # Create the resource configuration
+    my @sedoptions = undef;
+    push @sedoptions, "-e 's|%SID%|$sid|g'";
+    push @sedoptions, "-e 's|%HDB_INSTANCE%|$instance_id|g'";
+    my $automated_register = get_required_var('AUTOMATED_REGISTER');
+    my ($virtual_ip, $virtual_netmask) = split '/', get_required_var('VIRTUAL_IP_CIDR');
+    push @sedoptions, "-e 's|%AUTOMATED_REGISTER%|$automated_register|g'";
+    push @sedoptions, "-e 's|%VIRTUAL_IP_ADDRESS%|$virtual_ip|g'";
+    push @sedoptions, "-e 's|%VIRTUAL_IP_NETMASK%|$virtual_netmask|g'";
+
+    # We need to execute the sed command node by node
+    my $cmd          = undef;
+    my $cluster_conf = 'hana_cluster.conf';
+    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$cluster_conf -o /tmp/$cluster_conf";
+    $cluster_conf = "/tmp/" . $cluster_conf;
+
+    # Avoid hostname change on reboot.  See bsc#1153402
+    assert_script_run "sed -i 's/DHCLIENT_SET_HOSTNAME=\"yes\"/DHCLIENT_SET_HOSTNAME=\"no\"/' /etc/sysconfig/network/dhcp";
```

Comment:
I'm guessing `lib/utils::file_content_replace()` can also be used here.

Reply:
No comments about this?

Reply:
> No comments about this?

Modified the code to use that function.